alibi_detect/utils/keops/kernels.py

```diff
@@ -114,3 +114,65 @@ def forward(self, x: LazyTensor, y: LazyTensor, infer_sigma: bool = False) -> La
         if len(dist.shape) < len(gamma.shape):
             kernel_mat = kernel_mat.sum(-1) / len(self.sigma)
         return kernel_mat
+
+
+class DeepKernel(nn.Module):
+    def __init__(
+        self,
+        proj: nn.Module,
+        kernel_a: nn.Module = GaussianRBF(trainable=True),
+        kernel_b: Optional[nn.Module] = GaussianRBF(trainable=True),
+        eps: Union[float, str] = 'trainable'
+    ) -> None:
+        """
```

Comment:
One for a follow-up - we need consistency for where we put docstrings, e.g. the `torch` one has the docstring in the class body not in the constructor which makes it hard to do side-by-side comparison. Please open an issue and delegate someone.